[JEP-237] Introduce initial documentation for FIPS-140 (#6675)
jenkins-infra/jenkins.io: content/doc/book/managing/system-properties.adoc

    Setting this to `true` is unsafe.
    See link:/redirect/class-filter/[documentation].

- name: jenkins.security.FIPS140.COMPLIANCE
  tags:
  - feature
  - security
  def: |
    `false`
  since: 2.424
  description: |

>     Whether Jenkins and plugins system should prefer link:https://csrc.nist.gov/pubs/fips/140-2/upd2/final[FIPS-140] compliant cryptography.

Reviewer: ```suggestion
    The use of link:https://csrc.nist.gov/pubs/fips/140-2/upd2/final[FIPS-140] compliant cryptography for Jenkins and plugins systems.
```

Author: the suggestion does not correlate with what this flag does.  It does not cause any change, it is merely there for code to look at to decide if it wants to change (not all code will look at this flag) - so implying that this uses compliant cryptography would be incorrect and something that we should steer away from.

Reviewer: Revised my comment.

>     Not all features/plugins have been adapted, and this indicates a preference only, merely setting this flag to `true` does not make Jenkins and its plugins FIPS-140 compliant see <<system-administraton/FIPS-140#,Jenkins and FIPS-140 Overview>> for more information.

Reviewer: ```suggestion
    Not all features/plugins have been adapted, and this only indicates a preference. If you set this flag to `true`, it does not make Jenkins and its plugins FIPS-140 compliant. Refer to <<system-administraton/FIPS-140#,Jenkins and FIPS-140 Overview>> for more information.
```

Author: ```suggestion
    Not all features/plugins have been adapted, and this only indicates a preference.
    If you set this flag to `true`, it does not make Jenkins and its plugins FIPS-140 compliant. 
    Refer to <<system-administraton/FIPS-140#,Jenkins and FIPS-140 Overview>> for more information.
```

    Setting this to `true` is unsafe.
    See link:/redirect/class-filter/[documentation].

- name: jenkins.security.FIPS140.COMPLIANCE
  tags:
  - feature
  - security
  def: |
    `false`
  since: 2.424
  description: |
    If Jenkins and plugins systems should prefer link:https://csrc.nist.gov/pubs/fips/140-2/upd2/final[FIPS-140] compliant cryptography.
    Not all features/plugins have been adapted, and this only indicates a preference.
    If you set this flag to `true`, it does not make Jenkins and its plugins FIPS-140 compliant. 

>     Refer to <<system-administraton/FIPS-140#,Jenkins and FIPS-140 Overview>> for more information.

Author: ```suggestion
    Refer to <<../system-administraton/fips-140#,Jenkins and FIPS-140 Overview>> for more information.
```
